tasks/Task.cpp

```diff
@@ -1082,27 +1091,35 @@ void Task::balanceConfiguration(Arena::IDevice& device)
         BalanceWhiteAuto::BALANCE_WHITE_AUTO_OFF) {
         GenApi::CFloatPtr value = device.GetNodeMap()->GetNode("BalanceRatio");
 
-        LOG_INFO_S << "Setting BalanceRatioSelector to Blue";
-        Arena::SetNodeValue<gcstring>(device.GetNodeMap(),
-            "BalanceRatioSelector",
-            "Blue");
+        if (!base::isUnknown(_analog_controller_config.get().blue_balance_ratio)) {
+            LOG_INFO_S << "Setting BalanceRatioSelector to Blue";
+            Arena::SetNodeValue<gcstring>(device.GetNodeMap(),
+                "BalanceRatioSelector",
+                "Blue");
 
-        LOG_INFO_S << "Setting Blue Balance Ratio Value";
-        value->SetValue(_analog_controller_config.get().blue_balance_ratio);
+            LOG_INFO_S << "Setting Blue Balance Ratio Value";
+            value->SetValue(_analog_controller_config.get().blue_balance_ratio);
+        }
 
-        LOG_INFO_S << "Setting BalanceRatioSelector to Green";
-        Arena::SetNodeValue<gcstring>(device.GetNodeMap(),
-            "BalanceRatioSelector",
-            "Green");
+        if (!base::isUnknown(_analog_controller_config.get().green_balance_ratio)) {
```

Comment:
I really recommend you to save the property value in a variable instead of doing many gets